augur/curate/titlecase.py

```diff
@@ -93,7 +93,7 @@ def run(args, records):
 
         for field in args.titlecase_fields:
             titlecased_string = titlecase(record.get(field, ""), articles, abbreviations)
-            field_value = record.get(field, "")
+            field_value = record.get(field, "Not a field")
```

Comment:
Using strings like this `"Not a field"` as placeholders can be confusing. For example, in the messages below which use `{field_value!r}` the placeholder value will represented as if it was the actual value found in the record. Let's fix this while addressing the missing/null-valued field behaviour.

Reply:
Agree with you that the placeholder can be confusing. Just documenting my thought process that this was a hacky way to get a descriptive error message below.

```
  $ echo '{"region":"europe", "country":"france" }' \
  >   | ${AUGUR} curate titlecase --titlecase-fields "region" "country" "division" "location" "not exist"
  ERROR: Failed to titlecase 'division':'Not a field' in record 0
  [2]
```

Thanks, I'll revisit this and change it

Reply:
Clarification question about:

> I'd think to make non-existent and existent-but-null-valued fields pass thru unchanged by default.

The original script ([transform-string-fields](https://github.com/nextstrain/monkeypox/blob/master/ingest/bin/transform-string-fields)) allows non-existent fields to get passed through and added to the final entry.

```
  $ echo '{"region":"europe", "country":"france" }' \
  >   | ${AUGUR} curate titlecase --titlecase-fields "region" "country" "division" "location" "not exist"
  {"region": "Europe", "country": "France", "division": "", "location": "", "not exist": ""}
```

Just making sure this is what you mean by "non-existent pass thru unchanged by default". Or were you expecting:

```
  $ echo '{"region":"europe", "country":"france" }' \
  >   | ${AUGUR} curate titlecase --titlecase-fields "region" "country" "division" "location" "not exist"
  {"region": "Europe", "country": "France"}
```

I'm adding these as test cases to make sure we have documented edgecases.

Reply:
I'd expect the latter, i.e. fields which didn't exist in an input record are _not_ added to output records. Something like:

```
  $ echo '{"a": "hello world", "b": null}' \
  >   | ${AUGUR} curate titlecase --titlecase-fields a b c
  {"a": "Hello World", "b": null}
```